src/UBAFeeCalculator/UBAFeeSpokeCalculatorAnalog.test.ts

```diff
@@ -0,0 +1,167 @@
+import { BigNumber } from "ethers";
+import { getEventFee, getDepositFee, getRefundFee } from "./UBAFeeSpokeCalculatorAnalog";
+import { toBN } from "../utils";
+import { MockUBAConfig } from "../clients/mocks";
+import { computePiecewiseLinearFunction } from "./UBAFeeUtility";
+
+describe("UBAFeeSpokeCalculatorAnalog", () => {
+  const defaultConfig = new MockUBAConfig();
+
+  describe("getEventFee", () => {
+    it("should calculate the balancing fee for an inflow event", () => {
+      const amount = BigNumber.from(10);
+      const lastRunningBalance = BigNumber.from(1000);
+      const lastIncentiveBalance = BigNumber.from(1000);
+      const chainId = 1;
+      const flowType = "inflow";
+      const fee = getEventFee(amount, flowType, lastRunningBalance, lastIncentiveBalance, chainId, defaultConfig);
+      expect(fee.balancingFee.toString()).toEqual("20");
+    });
+
+    it("should calculate the balancing fee for an outflow event", () => {
+      const amount = BigNumber.from(10);
+      const lastRunningBalance = BigNumber.from(1000);
+      const lastIncentiveBalance = BigNumber.from(1000);
+
+      const chainId = 1;
+      const flowType = "outflow";
+      const fee = getEventFee(amount, flowType, lastRunningBalance, lastIncentiveBalance, chainId, defaultConfig);
+      expect(fee.balancingFee.toString()).toEqual("0");
+    });
+
+    it("should return a balanceFee of 0 if the amount is 0", () => {
+      const amount = BigNumber.from(0);
+      const lastRunningBalance = BigNumber.from(1000);
+      const lastIncentiveBalance = BigNumber.from(1000);
+      const chainId = 1;
+      const flowType = "outflow";
+      const fee = getEventFee(amount, flowType, lastRunningBalance, lastIncentiveBalance, chainId, defaultConfig);
+      expect(fee.balancingFee.toString()).toEqual("0");
+    });
+
+    it("should return a balance fee 0 if lastIncentiveBalance is negative", () => {
+      const amount = BigNumber.from(10);
+      const lastRunningBalance = BigNumber.from(1000);
+      const lastIncentiveBalance = BigNumber.from(-1000);
+      const chainId = 1;
+      const flowType = "outflow";
+      const fee = getEventFee(amount, flowType, lastRunningBalance, lastIncentiveBalance, chainId, defaultConfig);
+      expect(fee.balancingFee.toString()).toEqual("0");
+    });
+
+    it("should return a balance fee 0 if lastIncentiveBalance is 0", () => {
+      const amount = BigNumber.from(10);
+      const lastRunningBalance = BigNumber.from(1000);
+      const lastIncentiveBalance = BigNumber.from(0);
+      const chainId = 1;
+      const flowType = "outflow";
+      const fee = getEventFee(amount, flowType, lastRunningBalance, lastIncentiveBalance, chainId, defaultConfig);
+      expect(fee.balancingFee.toString()).toEqual("0");
+    });
+
+    describe("getEventFee should correctly apply a reward multiplier", () => {
+      for (const multiplier of [-23, 0, 23]) {
```

Comment:
These multipliers should be in `wei` so rather than `-23` do `-toBNWei("23")`

Reply:
I'd also like to see some test values with other numbers like:
- `toBNWei("5.5")`
- `toBNWei("0.2")`
- `toBNWei("1.2")` etc.

Reply:
Yep! That's now added in: 
```
      for (const rawMultiplier of ["-1", "1.2", "0.2", "1.2", "3.4", "7"]) {
        const multiplier = utils.parseEther(rawMultiplier);
```